cangen.c

```diff
@@ -504,7 +504,7 @@ int main(int argc, char **argv)
 			}
 			if (polltimeout) {
 				/* wait for the write socket (with timeout) */
-				if (poll(&fds, 1, polltimeout) < 0) {
+				if (poll(&fds, 1, polltimeout) <= 0) {
```

Comment:
Can you fix the `-EINTR` here, so that it's not treated as an error.

Reply:
I have add an extra patch.